app/controllers/admin/subscriptions_controller.rb

```diff
@@ -52,10 +52,30 @@ def unlink_course
       redirect_back fallback_location: root_path, notice: 'Cours retiré avec succès !'
     end
 
+    def confirm
+      subscription = Subscription.find(params[:id])
+      if subscription.confirmed?
+        redirect_to admin_subscriptions_path, alert: "L'inscription est deja confirmée !"
```

Comment:
We need an I18n translation here. Also: `déjà`

Reply:
There is a bug, because this message do not appear when we try to confirm or archive a subscription that is already confirmed or archived. 
I      I    IIIIIII    II           IIIIIIIp
IIIIIII     IIII       II           IIIIIIId
I      I    IIIIIIII   IIIIIIIII    II